From eae35a3614efbedd09f2958c65f8004e5213d5bf Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 19 Mar 2019 14:16:59 +0530
Subject: [PATCH 118/767] octeontx2-pf: Keep link notification enabled until
 device detach

commit 09f9e9c4a84fde7efaf69ee824238fb8c7ce9ba1 from
git@git.assembla.com:cavium/WindRiver.linux.git

Keep link notification enabled until device detach,
so that link notification broadcasting to VFs are not broken.

Change-Id: Ie73440100c1f37e12689e241dab97a825405f563
Signed-off-by: Linu Cherian <lcherian@marvell.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index a80cfb92dd32..d3f067cde895 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -962,9 +962,6 @@ int otx2_stop(struct net_device *netdev)
 	otx2_rxtx_enable(pf, false);
 	otx2_destroy_ethtool_flows(pf);
 
-	/* Disable link notifications */
-	otx2_cgx_config_linkevents(pf, false);
-
 	pf->intf_down = true;
 	/* 'intf_down' may be checked on any cpu */
 	smp_wmb();
@@ -1291,6 +1288,10 @@ static void otx2_remove(struct pci_dev *pdev)
 		return;
 
 	pf = netdev_priv(netdev);
+
+	/* Disable link notifications */
+	otx2_cgx_config_linkevents(pf, false);
+
 	unregister_netdev(netdev);
 	otx2_destroy_ethtool_flows(pf);
 
-- 
2.31.1

